app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt

```diff
@@ -42,6 +45,21 @@ class SchedulingFragment : PreferenceFragmentCompat(),
         }
     }
 
+    override fun onDisplayPreferenceDialog(preference: Preference?) {
+        when (preference) {
+            is ListPreference -> {
+                val dialogFragment = M3ListPreference.newInstance(preference.getKey())
+                dialogFragment.setTargetFragment(this, 0)
+                dialogFragment.show(
+                    parentFragmentManager,
+                    M3ListPreference.PREFERENCE_DIALOG_FRAGMENT_TAG
+                )
+            }
+
+            else -> super.onDisplayPreferenceDialog(preference)
+        }
+    }
+
```

Comment:
Google is really not giving us an easier way to get an MD3 list preference?

Reply:
Nope, none of the preferences are ready for MD3 stuff even in the latest release of the library. In Aurora Store, we have a similar class for EditTextPreference too for MD3 style.